### ISSUE TITLE
Rematch should probably use `isequal` instead of `==` for floating-point types

### ISSUE THREAD

**Issue by gafter** [RelationalAI-oss/Rematch.jl#35]
This:
```julia
using Rematch
nan = 0.0 / 0.0
f(x) = @match x begin
    [a,a] => :same
    [a,b] => :different
    _ => :other
end
println(f([nan, nan]))
```
prints "different".

**Comment by jerlich** [comment]
https://discourse.julialang.org/t/various-equalities-of-nan/42649

**Comment by gafter** [comment]
@jerlich I assume you provided that link in support of the bug report?

FYI, there is a revision of this package at `https://github.com/JuliaServices/Rematch2.jl` that I am preparing as a new release of `https://github.com/JuliaServices/Match.jl`.  It addresses this any many other issues in both `Match.jl` and `Rematch.jl`.

**Comment by jerlich** [comment]
Yes. That discourse discussion explains why Nan don't always equal nan.
